moduleframework/mtf_scheduler.py

```diff
@@ -219,31 +222,17 @@ def __init__(self, args, unknown):
 
     def avocado_run(self):
         self.json_tmppath = tempfile.mktemp()
-        avocado_args = "--json {JSON_LOG}".format(
-            JSON_LOG=self.json_tmppath)
+        avocado_args = ["--json", self.json_tmppath]
         if self.args.xunit:
-            avocado_args += " --xunit {XUNIT} ".format(XUNIT=self.args.xunit)
-        avocadoAction = "avocado {ACTION} {AVOCADO_ARGS}".format(
-            ACTION=self.args.action, AVOCADO_ARGS=avocado_args)
-
-        # run avocado with right cmd arguments
-        bash = process.run("{AVOCADO} {a} {b}".format(
-            AVOCADO=avocadoAction, a=self.additionalAvocadoArg, b=" ".join(self.tests)), shell=True, ignore_status=True)
-        common.print_info(bash.stdout, bash.stderr)
-        common.print_debug("Command used: ", bash.command)
-        return bash.exit_status
-
-    def avocado_general(self):
+            avocado_args += ["--xunit", self.args.xunit]
+        return self.avocado_general(avocado_default_args=avocado_args)
+
+    def avocado_general(self, avocado_default_args=[]):
         # additional parameters
         # self.additionalAvocadoArg: its from cmd line, whats unknown to this tool
-        avocado_args = ""  # when needed => specify HERE your additional stuff
-        avocadoAction = "avocado {ACTION} {AVOCADO_ARGS}".format(
-            ACTION=self.args.action, AVOCADO_ARGS=avocado_args)
-        bash = process.run("{AVOCADO} {a} {b}".format(
-            AVOCADO=avocadoAction, a=self.additionalAvocadoArg, b=" ".join(self.tests)), shell=True, ignore_status=True)
-        common.print_info(bash.stdout, bash.stderr)
-        common.print_debug("Command used: ", bash.command)
-        return bash.exit_status
+        avocadoAction = [self.AVOCADO, self.args.action] + avocado_default_args
+        command = subprocess.check_call(avocadoAction + self.additionalAvocadoArg + self.tests)
+        return command
```

Comment:
This should be in exception like is mentioned here: https://docs.python.org/2/library/subprocess.html#subprocess.check_call

```
try:
  subprocess.check_call
except CalledProcessError as cpe
   return cpe.return_code
```

Reply:
you are right